### PR TITLE
Fix for  #705. 409 Conflict while adding a new rule 

### PR DESCRIPTION
**Pull request by danchom**
Signed-off-by: Yordan Mihaylov <j.mihailov@prosyst.com>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.rest/src/main/java/org/eclipse/smarthome/automation/rest/internal/RuleResource.java [hunk 1]
```diff
@@ -97,16 +97,22 @@ private EnrichedRuleDTO enrich(Rule rule) {
     @Consumes(MediaType.APPLICATION_JSON)
     @ApiOperation(value = "Creates a rule.")
     @ApiResponses(value = { @ApiResponse(code = 201, message = "Created"),
-            @ApiResponse(code = 409, message = "Creation of the rule is refused. Rule with the same UID already exists.") })
+            @ApiResponse(code = 409, message = "Creation of the rule is refused. Rule with the same UID already exists."),
+            @ApiResponse(code = 400, message = "Creation of the rule is refused. Missing required parameter.") })
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.rest/src/main/java/org/eclipse/smarthome/automation/rest/internal/RuleResource.java]: "Missing required parameter" - is this the only possible cause? What happens if people send an invalid JSON structure? I would expect a 400 as well.

Comment (danchom) [inline bundles/automation/org.eclipse.smarthome.automation.rest/src/main/java/org/eclipse/smarthome/automation/rest/internal/RuleResource.java]: We don't have checks for the structure of the json, We are only looking for the required parameters on specified position in the json. In that way the structure is a corner case of missing parameter.

Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.rest/src/main/java/org/eclipse/smarthome/automation/rest/internal/RuleResource.java]: Where in the code does it fail if you send an invalid JSON structure? I guess that happens in the Jersey deserialization already, right? What does Jersey respond in such a situation? Is it a 400 as well?

Comment (danchom) [inline bundles/automation/org.eclipse.smarthome.automation.rest/src/main/java/org/eclipse/smarthome/automation/rest/internal/RuleResource.java]: Jersey does not do any validation. It just try to fill RuleResource from passed JSON object. If the JSON is wrong and it does not contain any of the RuleResource properties it just doesn't fill them.
I've created rest request with  wrong JSON structure (I've change the trigger's configuration to be defined in "config" node instead of "configuration" node) and the result was a response 201 with created rule which has empty triggers configuration.

Comment (danchom) [inline bundles/automation/org.eclipse.smarthome.automation.rest/src/main/java/org/eclipse/smarthome/automation/rest/internal/RuleResource.java]: Jersey do validation for incorrect json format. If it is not correct (last  "}" is missing ) then the response code is 500 Server error

Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.rest/src/main/java/org/eclipse/smarthome/automation/rest/internal/RuleResource.java]: 500 actually isn't very nice. Do you know if there is a way to have that changed to e.g. 400? But this is not directly related to this PR.
